mautrix_telegram/web/provisioning/__init__.py

```diff
@@ -528,6 +531,36 @@ async def retry_takeout(self, request: web.Request) -> web.Response:
         user.takeout_retry_immediate.set()
         return web.json_response({}, status=200)
 
+    async def login_qr(self, request: web.Request) -> web.Response:
+        _, user, err = await self.get_user_request_info(request, websocket=True)
+        if err is not None:
+            return err
+
+        await user.ensure_started(even_if_no_session=True)
+        qr_login = QRLogin(user.client, ignored_ids=[])
+
+        ws = web.WebSocketResponse(protocols=["net.maunium.telegram.login"])
+        await ws.prepare(request)
+
+        user_info = None
+        try:
+            await qr_login.recreate()
+            await ws.send_json({"code": qr_login.url, "timeout": 30})
+            user_info = await qr_login.wait()
+        except asyncio.TimeoutError:
+            await ws.send_json({"success": False, "error": "timeout"})
+            await ws.close()
```

Comment:
It should automatically refresh the QR at least a few times here instead of just closing immediately

Reply:
I left out the refresh logic just because Desktop reopens a new websocket on timeout

Reply:
Client side retries are probably a bad idea (they're completely wrong on whatsapp, DES-8281). It's better if the bridge provides a reasonable amount of QR codes automatically and the user has to manually click to refresh after they run out.